### DISCUSSION TITLE
How can I list LFS deleted files only?

### DISCUSSION THREAD

**Discussion by mathisc** [git-lfs/git-lfs#4668] [Q&A] [answered]
Hi all,

Let say I have a folder that is tracked by LFS. Couple of commits later I realize I do not need this folder anymore and simple do `git rm -rf <my_folder>`. How can I list all LFS files referenced in my repo but that are not used anymore? 
I'd like to use this list as an entry for `git filter repo` and reclaim some storage space in some gitlab projects that have a lot of (old) testing data stored in LFS : https://docs.gitlab.com/ee/user/project/repository/reducing_the_repo_size_using_git.html#repository-cleanup

Thanks for your help ;)

**Comment by chrisd8088** [comment, marked as answer]
I don't know if this will help, and there may well be a more elegant approach than what I've jotted down here.  But one option might be to compare the output of `git lfs ls-files` and `git lfs ls-files --deleted`, where the latter includes deleted files while the former does not.

For instance, with a simple test case where I created LFS files in a directory and then removed that directory (and committed both changes independently), I found this output just the deleted files.  Note that this does use the Unix command `comm` and Bash process substitution:
```
$ git init test
$ cd test
$ git lfs track '*.bin'
$ mkdir aaa zzz
$ echo aaa >aaa/aaa.bin
$ echo zzz >zzz/zzz.bin
$ git add .gitattributes aaa zzz
$ git commit -m init
$ echo foo >foo.bin
$ echo bar >bar.bin
$ git add foo.bin bar.bin
$ git commit -m foobar

$ git rm -r aaa
$ git commit -m noaaa
$ git rm -r zzz
$ git commit -m nozzz

$ git lfs ls-files
7d865e959b * bar.bin
b5bb9d8014 * foo.bin

$ git lfs ls-files --deleted
7d865e959b * bar.bin
b5bb9d8014 * foo.bin
72d4df2c38 - zzz/zzz.bin
17e682f060 - aaa/aaa.bin

$ comm -23 <(git lfs ls-files --deleted | sort) <(git lfs ls-files | sort)
17e682f060 - aaa/aaa.bin
72d4df2c38 - zzz/zzz.bin
```

You might experiment with some of the other options and arguments to `git lfs ls-files`, such as `--all` which inspects the entire history of the repository rather than just `HEAD`.

**Reply by mathisc** [reply]
Thanks a lot ;)
